keras_nlp/metrics/edit_distance.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Edit Distance metric."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class EditDistance(keras.metrics.Metric):
+    """Edit Distance metric.
+
+    This class implements the edit distance metric, sometimes called
+    Levenshtein Distance, as a `keras.metrics.Metric`. Essentially, edit
+    distance is the least number of operations required to convert one string to
+    another, where an operation can be one of substitution, deletion or
+    insertion. By default, this metric will compute the normalized score, where
+    the above score is divided by the number of tokens in the reference text.
+
+    This class can be used to compute character error rate (CER) and word error
+    rate (WER). You simply have to pass the appropriate tokenized text, and set
+    `normalize` to True.
+
+    Note on input shapes:
+    `y_true` and `y_pred` can either be tensors of rank 1 or ragged tensors of
```

Comment:
It can also be list?

Reply:
Yep, but I think we've stuck to specifying TensorFlow data structures in all other layers. For example, https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/tokenizers/sentence_piece_tokenizer.py#L35-L40 and https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/metrics/rouge_l.py#L29-L31.